test.sh

```diff
@@ -0,0 +1,27 @@
+#!/bin/sh
+: "${CMD:=$(realpath per)}"
+TESTDIR=$(mktemp -d)
+trap "rm -rf $TESTDIR" 0
+cd $TESTDIR
+
+# testing "args" "expected", green if passed, red if failed
+testing () {
+  $CMD $1 > got || echo "WARNING: $1 exited $?"
+  printf "$2\n" > wanted
```

Comment:
I chose this over the standard `printf "%s" "$VARIABLE"` to take advantage of escape sequence processing (i.e. '\n' to signal a newline).

Hmm... POSIX 2024 added the `$'...'` feature, but not `$"..."`, Is there any safe way to do this in pure POSIX shell?

Reply:
That's fair. I think that it's a bit too early to rely on the POSIX 2024 standard.

If we only need to handle newlines this way, we can just use multiline strings instead of the `\n` escape code.

Reply:
Commit e2d933a, multi-line strings